Spigot-API-Patches/0237-More-World-API.patch

```diff
@@ -0,0 +1,156 @@
++     */
++    @Nullable
++    Location locateNearestBiome(@NotNull Location origin, @NotNull Biome biome, int radius);
++
++    /**
++     * Locates the nearest biome based on an origin, biome type, and radius to search
++     * and step
++     *
++     * @param origin Origin location
++     * @param biome Biome to find
++     * @param radius radius to search
++     * @param step Search step 1 would mean checking every block, 8 -> every 8th block
++     * @return Location of biome or null if not found in specified radius
++     */
++    @Nullable
++    Location locateNearestBiome(@NotNull Location origin, @NotNull Biome biome, int radius, int step);
++
++    /**
++     * Checks if the world:
++     * <ul>
++     *     <li>evaporates water</li>
++     *     <li>dries sponges</li>
++     *     <li>has lava spread faster and further</li>
++     * </ul>
++     *
++     * @return true if ultrawarm, false if not
++     */
++    boolean isUltrawarm();
++
++    /**
++     * Checks if the world is natural.
++     * <p>
++     *     If {@code false}, compasses will spin randomly in the world.
++     *      * if {@code true}, nether portals will spawn zombified piglins.
++     * </p>
++     *
++     * @return true or false
++     */
++    boolean isNatural();
++
++    /**
++     * Gets the coordinate scaling of this world.
++     *
++     * @return the coordinate scale
++     */
++    double getCoordinateScale();
++
++    /**
++     * Checks if the world has skylight access
++     *
++     * @return whether there is skylight
++     */
++    boolean hasSkylight();
++
++    /**
++     * Checks if the world has a bedrock ceiling
++     *
++     * @return whether the world has a bedrock ceiling
++     */
++    boolean hasBedrockCeiling();
++
++    /**
++     * Checks if piglins will turn into Zombified Piglins in this world
++     *
++     * @return whether Piglins will <i>not</i> transform
++     */
++    boolean isPiglinSafe();
++
++    /**
++     * Checks if beds work
++     *
++     * @return whether beds work
++     */
++    boolean doesBedWork();
++
++    /**
++     * Checks if respawn anchors work
++     *
++     * @return whether respawn anchors work
++     */
++    boolean doesRespawnAnchorWork();
++
++    /**
++     * Checks if this world supports raids
++     *
++     * @return whether this world supports raids
++     */
++    boolean hasRaids();
++
++    /**
++     * Checks if this world has a fixed time
++     *
++     * @return whether this world has fixed time
++     */
++    boolean isFixedTime();
++
++    /**
++     * Gets the collection of materials that burn infinitely in this world.
++     *
++     * @return the materials that will forever stay lit by fire
++     */
++    @NotNull
++    Collection<Material> getInfiniburn();
++
++    /**
++     * Broadcast a message to all players
++     * in this world
++     *
++     * @param message Message to broadcast
++     */
++    void broadcastMessage(@NotNull String message);
```

Comment:
These will end up deprecated as part of #4842 - is it worth even adding them now?

Reply:
Yeah, you're right. I'll just leave a note to add the Component method after adventure get merged